locallib.php

```diff
@@ -166,13 +175,15 @@ public function save_settings(stdClass $data) {
         global $CFG;
         require_once($CFG->dirroot . '/mod/assign/submission/maharaws/lib.php');
 
-        $this->set_config('url', $data->assignsubmission_maharaws_url);
-        $this->set_config('key', $data->assignsubmission_maharaws_key);
-        $this->set_config('secret', $data->assignsubmission_maharaws_secret);
-        $this->set_config('debug', false);
-        $this->set_config('remoteuser', false);
-        $this->set_config('lock', $data->assignsubmission_maharaws_lockpages);
-        $this->set_config('username_attribute', 'email');
+        if ($this->can_configure()) {
+            $this->set_config('url', $data->assignsubmission_maharaws_url);
+            $this->set_config('key', $data->assignsubmission_maharaws_key);
+            $this->set_config('secret', $data->assignsubmission_maharaws_secret);
+            $this->set_config('debug', false);
+            $this->set_config('remoteuser', false);
+            $this->set_config('lock', $data->assignsubmission_maharaws_lockpages);
+            $this->set_config('username_attribute', 'email');
+        }
 
         // Test Mahara connection.
         try {
```

Comment:
it shows error if can't connect to the mahara instance. So I guess we should leave as is  this block. So the check will happen regardless cap check and if it's configured incorrectly it will show the errors. We are not changing this functionality.

Reply:
Thank you @dmitriim 

You're right. This makes sense to me.

Thanks,
Anu